### PR TITLE
topology2: intel: add generic Bluetooth offload pipelines

### PR DESCRIPTION
**Pull request by kv2019i**
Add generic Bluetooth pipeline definitions that matches what we've had with topology1. Augment passthrough pipelines to allow use of Bluetooth.

### CODE REVIEW

File: tools/topology/topology2/cavs-nocodec-bt.conf [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+<route.conf>
+<common_definitions.conf>
+<copier.conf>
+<pipeline.conf>
+<dai.conf>
+<host.conf>
+<dmic-default.conf>
+
+<bt-default.conf>
+<bt-generic.conf>
```
[the remark this answers is not in the document]
Comment (kv2019i) [reply tools/topology/topology2/cavs-nocodec-bt.conf:33]: @ranj063 I considered that, but given BT is a pain to test already and nocodec is test ground for complex mixer usage, it seems way more simple to have a simple passthrough tplg for BT, with no other complexities in the way.

**Review by ranj063 — APPROVED** [review]

File: tools/topology/topology2/cavs-nocodec-bt.conf (hunk 1 above)
[the remark this answers is not in the document]
Comment (ranj063) [reply tools/topology/topology2/cavs-nocodec-bt.conf:33]: ok fair enough

**Comment by kv2019i** [comment]
V2 (Jan 23rd) update:
- removed changes to generic passthrough pipeline definitions, override the values in bt-generic.conf instead
- fix A2DP profile support (48kHz/stereo) -> in V1, the FE->BE connection converted samples incorrectly S32_LE .. the SSP driver stripped the LSB bits so no runtime error was raised, but this is still wrong

**Comment by kv2019i** [comment]
@ranj063 I probably should fix this before merge to align with #6627 ... right?

**Comment by ranj063** [comment]
> @ranj063 I probably should fix this before merge to align with #6627 ... right?

@kv2019i yes thats right. We should

**Comment by marc-hb** [comment]
Following the long and winded road starting from https://sof-ci.01.org/sofpr/PR6971/build3487/devicetest/index.html (internal ID 20119), I found that the reason why ADLP_RVP_SDW_IPC4ZPH failed is as usual NOT because that model was "Not Available" but because:

```
09:02:56  [2023-01-23 17:02:55 UTC][SOF-SH] Now time: 2023-01-23T17:02:55+00:00 | Device time: 2023-01-23T17:02:55+00:00
09:02:57  [2023-01-23 17:02:56 UTC][SOF-SH] ERROR: Failed to retrieve Topology resource from
       [http://sof-storage.ostc.intel.com/pr/sof/PR6971-3487/build3487/tplg/topology2/avs-tplg/sof-adl-rt711-4ch.tplg](http://sof-storage.ostc.intel.com/pr/sof/PR6971-3487/build3487/tplg/topology2/avs-tplg/sof-adl-rt711-4ch.tplg
```
(/job/testflow/62327)

Could this be caused by this PR?

**Comment by kv2019i** [comment]
V3 (Jan 24th) update:
- aligned refactoring done in #6627 to this one as well
- fixed rebase problem (thanks @marc-hb and @juimonen ) that unintentionally removed one tplg

**Review by juimonen — APPROVED** [review]
Pierre's typo patch went upstream, so not sure if needed to mention that in commit message. LGTM.

**Comment by plbossart** [comment]
> Pierre's typo patch went upstream, so not sure if needed to mention that in commit message. LGTM.

it does impact the alsa-plugin tools one would think, so the docker image should be updated, no? @kv2019i @juimonen @fredoh9

**Comment by juimonen** [comment]
> > Pierre's typo patch went upstream, so not sure if needed to mention that in commit message. LGTM.
> 
> it does impact the alsa-plugin tools one would think, so the docker image should be updated, no? @kv2019i @juimonen @fredoh9

yes definitely, need update to latest upstream. Just meant that Kai's original message was there perhaps to inform that the needed patch is not yet in upstream alsa-utils head. But now it is -> so we need update also.

**Review by lgirdwood — APPROVED** [review]

**Comment by kv2019i** [comment]
Let's proceed with merge.

**Comment by marc-hb** [comment]
There was "logical" conflict between this 4476171620b63928 and ae8ca22a590d3. They both passed separately but not together - and CI never tested them together.

This sort of "logical conflict" is very rare and too hard to catch in CI, not worth the effort. The mitigations are:
- Do an empty `git commit --amend -C HEAD` and force-push when the PR hasn't been tested for a while. Don't rebase as it makes it extremely difficult to see what changed.
- Keep an eye on the git log and communicate more! :-)

**Pull request merged**